### PR TITLE
Feature/setup param

### PR DESCRIPTION
**Pull request by AHReccese**
#### Reference Issues/PRs
#12 
#19

### CODE REVIEW

File: reserver/reserver_param.py [hunk 1]
```diff
@@ -21,4 +21,9 @@
     "url": r'^(http|https)://[a-zA-Z0-9.-_]+\.[a-zA-Z]{2,}(/\S*)?$',
 }
 INVALID_PACKAGE_PARAMETER_NAME_ERROR = "Given parameter doesn't exist among the supported user allowed parameters."
-INVALID_PACKAGE_PARAMETER_VALUE_ERROR = "Invalid value for {parameter} that should be a valid {regex}"
\ No newline at end of file
+INVALID_PACKAGE_PARAMETER_VALUE_ERROR = "Invalid value for {parameter} that should be a valid {regex}"
+INVALID_CONFIG_FILE_NAME_ERROR = "Given file name for user-defined setup.py params is not a string."
```
Comment (sadrasabouri) [inline reserver/reserver_param.py:25]: Are you sure about `setup.py`?

File: reserver/reserver_param.py [hunk 2]
```diff
@@ -21,4 +21,9 @@
...
+PARAM_FILE_DOES_NOT_EXIST_ERROR = "Given file doesn't exist."
+INVALID_INPUT_USER_PARAM = "Invalid input for user params."
+UNEQUAL_PARAM_NAME_LENGTH_ERROR = "You should pass either one single file path to be used for setup.py parameters \
```
Comment (sadrasabouri) [inline reserver/reserver_param.py:28]: Not sure about `setup.py` again.
Also, I prefer this:
```
`user_params_path` should be either a path or a list of paths to the config file(s).
```

File: reserver/reserver_param.py [hunk 3]
```diff
@@ -21,4 +21,9 @@
...
+or per each package name, there should be a specific dedicated file path."
```
Comment (sadrasabouri) [inline reserver/reserver_param.py:29]: Also, add a new line at the end of this file.

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
I left some comments. Thank you for your efforts

File: tests/config.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+    "description": "[config] This name has been reserved using Reserver",
+    "author": "[config] Development Team",
+    "author_email": "configtest@test.com",
+    "url": "https://configurl.com",
+    "download_url": "https://configdownload_url.com",
+    "source": "https://configgithub.com/source",
+    "license": "[config] MIT"
+}
```
[the remark this answers is not in the document]
Comment (AHReccese) [reply tests/config.json:9]: after this PR, I will handle CLI in which we can handle custom batch uploads
then I will update `README.md`

File: reserver/reserver_param.py (hunk 1 above)
Comment (AHReccese) [reply reserver/reserver_param.py]: I update the error message, changing `setup.py` with `package`.

File: reserver/reserver_param.py (hunk 2 above)
Comment (AHReccese) [reply reserver/reserver_param.py]: I update the error message, changing `setup.py` with `package`.

**Review by sadrasabouri — APPROVED** [review]

**Pull request merged**